lib/sqlalchemy/sql/elements.py

```diff
@@ -4122,7 +4238,7 @@ def over(self, partition_by=None, order_by=None, range_=None, rows=None):
         )
 
     @util.memoized_property
-    def type(self):
+    def type(self) -> TypeEngine[_T]:
         wgt = self.element.within_group_type(self)
         if wgt is not None:
             return wgt
```

Comment:
Calls an external function `within_group_type()` which returns `None` by default, in which case normal `type` is returned; otherwise `WithinGroup.type()` has to return the result of `within_group_type()`, which is unknown - it is probably meant to be overriden by some other classes that require such behaviour. That's why mypy complains about returning `Any` here - guessing I have to just `# type: ignore` that.